modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpConnectionManager.java

```diff
@@ -194,46 +197,55 @@ public HttpClientConnector getHTTPHttpClientConnector(String scheme, BConnector
         }
 
         BStruct options = (BStruct) bConnector.getRefField(OPTIONS_STRUCT_INDEX);
-        BStruct ssl = (BStruct) options.getRefField(SSL_STRUCT_INDEX);
-        BStruct followRedirects = (BStruct) options.getRefField(FOLLOW_REDIRECT_STRUCT_INDEX);
-        int followRedirect = followRedirects.getBooleanField(FOLLOW_REDIRECT_INDEX);
-        Long maxRedirectCount = followRedirects.getIntField(MAX_REDIRECT_COUNT);
-        String trustStoreFile = ssl.getStringField(TRUST_STORE_FILE_INDEX);
-        String trustStorePassword = ssl.getStringField(TRUST_STORE_PASSWORD_INDEX);
-        String keyStoreFile = ssl.getStringField(KEY_STORE_FILE_INDEX);
-        String keyStorePassword = ssl.getStringField(KEY_STORE_PASSWORD_INDEX);
-        String sslEnabledProtocols = ssl.getStringField(SSL_ENABLED_PROTOCOLS_INDEX);
-        String ciphers = ssl.getStringField(CIPHERS_INDEX);
-        String sslProtocol = ssl.getStringField(SSL_PROTOCOL_INDEX);
-
-        senderConfiguration.setFollowRedirect(followRedirect == 1 ? true : false);
-        senderConfiguration.setMaxRedirectCount(maxRedirectCount.intValue());
-        if (StringUtils.isNotBlank(trustStoreFile)) {
-            senderConfiguration.setTrustStoreFile(trustStoreFile);
-        }
-        if (StringUtils.isNotBlank(trustStorePassword)) {
-            senderConfiguration.setTrustStorePass(trustStorePassword);
-        }
-        if (StringUtils.isNotBlank(keyStoreFile)) {
-            senderConfiguration.setKeyStoreFile(keyStoreFile);
-        }
-        if (StringUtils.isNotBlank(keyStorePassword)) {
-            senderConfiguration.setKeyStorePassword(keyStorePassword);
+        if (options == null) {
+            return httpConnectorFactory.createHttpClientConnector(properties, senderConfiguration);
```

Comment:
A better approach to do this would be, create a new method like "populateSenderConfigurationOptions(SenderConfiguration senderConfiguration, BStruct options) and put the options populating logic inside that. And call this method from this method when options are not null.

Reply:
Updated the code with the requested change